consensus/quorum/one-node-staked-vote.go

```diff
@@ -0,0 +1,104 @@
+package quorum
+
+import (
+	"math/big"
+
+	"github.com/harmony-one/bls/ffi/go/bls"
+	"github.com/harmony-one/harmony/internal/common"
+	"github.com/harmony-one/harmony/numeric"
+	"github.com/harmony-one/harmony/staking/effective"
+)
+
+var (
+	twoThirds = numeric.NewDec(2).QuoInt64(3)
+)
+
+type stakedVoter struct {
+	isActive, isHarmonyNode bool
+	effective               numeric.Dec
+}
+
+type stakedVoteWeight struct {
+	SignatureReader
+	// EPOS based staking
+	validatorStakes            map[*bls.PublicKey]stakedVoter
```

Comment:
Is mapping by *bls.PublicKey safe? How about mapping by the bytes of the key?

Reply:
It should be safe, if concerned about pointer usage then invariant of type safety in code should be maintained because all the places that added a PublicKey used bls.PublicKey which would .Deserialize as the bytes, so I moved it up one level and let the called functions that operate with quorum handle the call to .Deserialize instead of prior to call it